src/components/settings-menu/BackupSection/WalletSelectionView.js

```diff
@@ -0,0 +1,189 @@
+  weight: 'regular',
+})``;
+
+const CheckmarkIcon = styled(Icon).attrs({
+  name: 'checkmarkCircled',
+})`
+  align-self: center;
+  margin-bottom: 1px;
+  margin-right: 7px;
+`;
+
+const GreenCheck = styled(CheckmarkIcon).attrs({
+  color: colors.green,
+})`
+  box-shadow: 0px 4px 6px ${colors.alpha(colors.green, 0.4)};
+`;
+
+const GreyCheck = styled(CheckmarkIcon).attrs({
+  color: colors.blueGreyDark50,
+})`
+  box-shadow: 0px 4px 6px ${colors.alpha(colors.blueGreyDark50, 0.4)};
+`;
+
+const WarningIcon = styled(Icon).attrs({
+  color: colors.orangeLight,
+  name: 'warning',
+})`
+  align-self: center;
+  box-shadow: 0px 4px 6px ${colors.alpha(colors.orangeLight, 0.4)};
+  margin-right: 7px;
+`;
+
+const WalletSelectionView = () => {
+  const { navigate } = useNavigation();
+  const { walletNames, wallets } = useWallets();
+  const onPress = useCallback(
+    (walletId, name) => {
+      const wallet = wallets[walletId];
+      if (wallet.backedUp || wallet.imported) {
+        navigate('SettingsBackupView', {
+          imported: wallet.imported,
+          title: name,
+          type: 'AlreadyBackedUpView',
+          walletId,
+        });
+      } else {
+        navigate('SettingsBackupView', {
+          title: name,
+          type: 'NeedsBackupView',
+          walletId,
+        });
+      }
+    },
+    [navigate, wallets]
+  );
+
+  return Object.keys(wallets)
+    .filter(key => wallets[key].type !== WalletTypes.readOnly)
+    .map(key => {
+      const wallet = wallets[key];
+      const visibleAccounts = wallet.addresses.filter(a => a.visible);
+      const account = visibleAccounts[0];
+      const totalAccounts = visibleAccounts.length;
+      const { color, label, index, address } = account;
+      let labelOrName = label;
+      if (!label) {
+        if (walletNames[address]) {
+          labelOrName = walletNames[address];
+        }
+      }
+
+      return (
+        <Column key={key}>
+          <ButtonPressAnimation
+            onPress={() =>
+              onPress(key, label || abbreviations.address(address, 4, 6))
+            }
+            scaleTo={0.98}
+          >
+            <Row height={56}>
+              <Row alignSelf="center" flex={1} marginLeft={15}>
+                <ContactAvatar
+                  alignSelf="center"
+                  color={color}
+                  marginRight={10}
+                  size="smedium"
+                  value={labelOrName || `${index + 1}`}
+                />
+                <ColumnWithMargins margin={3} marginBottom={0.5}>
+                  <Row>
+                    {labelOrName ? (
+                      <AccountLabel>{labelOrName}</AccountLabel>
+                    ) : (
+                      <Address address={address} />
+                    )}
+                  </Row>
+                  {totalAccounts > 1 ? (
+                    <BottomRowText weight={fonts.weight.medium}>
+                      And {totalAccounts - 1} more wallets
```

Comment:
Should toggle between "more wallet" vs "more wallets" (i.e, if totalAccounts is 2)